# Viết code Python đầu tiên của bạn

Thời lượng ước tính: **25** phút

## Mục tiêu

Sau khi làm lab này, học viên sẽ có thể:

- Viết code cơ bản trong Python
- Làm việc với nhiều kiểu dữ liệu khác nhau trong Python
- Chuyển đổi dữ liệu từ kiểu này sang kiểu khác
- Sử dụng các biểu thức và biến để thực hiện các phép toán


<h2>Mục lục</h2>
<div class="alert alert-block alert-info" style="margin-top: 20px">
    <ul>
        <li>
            <a href="#hello">Nói "Hello" trong Python</a>
            <ul>
                <li><a href="version">Chúng ta đang dùng Python phiên bản nào?</a></li>
                <li><a href="comments">Viết chú thích trong Python</a></li>
                <li><a href="errors">Lỗi trong Python</a></li>
                <li><a href="python_error">Python có biết lỗi của bạn trước khi chạy code không?</a></li>
                <li><a href="exercise">Exercise: Chương trình đầu tiên</a></li>
            </ul>
        </li>
        <li>
            <a href="#types_objects">Các kiểu đối tượng trong Python</a>
            <ul>
                <li><a href="int">Số nguyên</a></li>
                <li><a href="float">Số thực</a></li>
                <li><a href="convert">Chuyển đổi một kiểu đối tượng này sang một kiểu đối tượng khác</a></li>
                <li><a href="bool">Kiểu dữ liệu Boolean</a></li>
                <li><a href="exer_type">Exercise: Các kiểu dữ liệu</a></li>
            </ul>
        </li>
        <li>
            <a href="#expressions">Biểu thức và biến</a>
            <ul>
                <li><a href="exp">Biểu thức</a></li>
                <li><a href="exer_exp">Exercise: Biểu thức</a></li>
                <li><a href="var">Biến</a></li>
                <li><a href="exer_exp_var">Exercise: Biểu thức và biến trong Python</a></li>
            </ul>
        </li>
    </ul>
    <p>
        Thời lượng ước tính: <strong>25 phút</strong>
    </p>
</div>

<hr>


### <h2 id="hello">Nói "Hello" với thế giới trong Python</h2>


Khi học một ngôn ngữ lập trình mới, theo thông lệ, bạn nên bắt đầu với ví dụ "hello world". Nói một cách đơn giản, một dòng code lập trình này sẽ đảm bảo rằng chúng ta biết cách in một chuỗi ở đầu ra và cách thực hiện code trong các cell trong notebook.


<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
[Tip]: Để thực thi code Python trong cell code bên dưới, hãy nhấp vào cell để chọn và nhấn <kbd>Shift</kbd> + <kbd>Enter</kbd>.
</div>
<hr/>


In [1]:
# Thử output Python đầu tiên

print('Hello, Python!')

Hello, Python!


Sau khi thực thi cell ở trên, bạn sẽ thấy Python in ra <code>Hello, Python!</code>. Chúc mừng bạn đã chạy code Python đầu tiên của mình!



<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
    [Tip:] <code>print()</code> là một hàm. Bạn truyền string <code>'Hello, Python!'</code> làm đối số để hướng dẫn những gì Python cần in.
</div>
<hr/>


<h3 id="version">Chúng ta đang dùng Python phiên bản nào?</h3>


<p>
    Có hai phiên bản phổ biến của ngôn ngữ lập trình Python được sử dụng hiện nay là Python 2 và Python 3. Cộng đồng Python đã quyết định chuyển từ Python 2 sang Python 3 và nhiều thư viện phổ biến đã thông báo rằng họ sẽ không hỗ trợ Python 2 nữa.
</p>
<p>
    Vì Python 3 là xu hướng của tương lai nên trong khóa học này, chúng tôi sẽ chỉ sử dụng nó. Làm cách nào để biết rằng notebook của chúng ta được thực thi bởi Python 3 runtime? Chúng ta có thể nhìn vào góc trên bên phải của notebook này và thấy "Python 3".
</p>
<p>
    Chúng ta cũng có thể hỏi trực tiếp Python và nhận được câu trả lời chi tiết. Hãy thử thực hiện code sau:
</p>


In [2]:
# Kiểm tra phiên bản Python

import sys
print(sys.version)

3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]


<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
    [Tip:] <code>sys</code> là một mô-đun tích hợp có chứa nhiều tham số và hàm dành riêng cho hệ thống, bao gồm cả phiên bản Python đang được sử dụng. Trước khi sử dụng, chúng ta phải <code>import</code> nó.
</div>
<hr/>


<h3 id="comments">Viết chú thích trong Python</h3>


<p>
    Ngoài việc viết code, hãy lưu ý rằng bạn luôn nên thêm chú thích vào code của mình. Nó sẽ giúp người khác hiểu điều bạn đang cố gắng hoàn thành (lý do tại sao bạn viết một đoạn code nhất định). Điều này không chỉ giúp <strong>người khác</strong> hiểu code của bạn, nó cũng giúp nhắc nhở <strong>bạn</strong> khi bạn quay trở lại sau vài tuần hoặc vài tháng.</p>

<p>
    Để viết chú thích bằng Python, hãy sử dụng ký hiệu số <code>#</code> trước khi viết chú thích. Khi bạn chạy code, Python sẽ bỏ qua mọi thứ sau <code>#</code> trên một dòng nhất định.
</p>


In [3]:
# Thực hành viết chú thích

print('Hello, Python!') # This line prints a string
# print('Hi')

Hello, Python!


<p>
    Sau khi thực thi cell ở trên, bạn sẽ nhận thấy rằng <code>This line prints a string</code> không xuất hiện ở output, vì đó là một chú thích (nên bị Python bỏ qua).
</p>
<p>
    Dòng thứ hai cũng không được thực thi vì <code> print('Hi')</code> cũng được đặt trước bởi kí hiệu số (<code>#</code>)! Vì đây không phải là một chú thích giải thích từ lập trình viên, mà là một dòng code thực tế, chúng ta có thể nói rằng lập trình viên đã <em>chú thích</em> dòng code thứ hai đó.
</p>


<h3 id="errors">Lỗi trong Python</h3>


<p>Ai cũng mắc sai lầm. Đối với nhiều loại lỗi, Python sẽ đưa ra thông báo để cho bạn biết rằng bạn đã mắc lỗi. Điều quan trọng là phải đọc các thông báo lỗi một cách cẩn thận để thực sự hiểu bạn đã mắc lỗi ở đâu và cách bạn có thể sửa nó.</p>
<p>Ví dụ, nếu bạn viết sai chính tả <code>print</code> thành <code>frint</code>, Python sẽ hiển thị thông báo lỗi. Bạn hãy thử xem:</p>



In [4]:
# In ra thông báo lỗi

frint("Hello, Python!")

NameError: ignored

<p>Thông báo lỗi cho bạn biết:  
<ol>
    <li>nơi nào có lỗi (hữu ích hơn trong các tập lệnh hoặc cell notebook lớn)</li>
    <li>đó là loại lỗi nào (Tên lỗi)</li>
</ol>
<p>Ở đây, Python đã cố gắng chạy hàm <code> frint</code>, nhưng không thể xác định <code> frint</code> là gì vì nó không phải là một hàm tích hợp và chúng ta cũng chưa xác định nó trước đó.</p>



<p>
   Bạn sẽ nhận thấy rằng nếu chúng ta mắc một loại lỗi khác, do quên đóng string, chúng ta sẽ gặp một lỗi khác (ví dụ, <code>SyntaxError</code>). Thử nó dưới đây:
</p>


In [6]:
# Hãy thử tạo một thông báo lỗi

print("Hello, Python!")

Hello, Python!


<h3 id="python_error">Python có nhận ra lỗi của bạn trước khi nó chạy code không?</h3>


Python được gọi là <em>ngôn ngữ thông dịch</em>. Ngôn ngữ biên dịch sẽ kiểm tra toàn bộ chương trình tại thời điểm biên dịch và có thể cảnh bảo bạn về toàn bộ lớp lỗi trước khi thực thi. Ngược lại, python thông dịch dòng tập lệnh khi nó thực thi.Pyhton sẽ ngừng thực thi toàn bộ chương trình khi nó gặp lỗi (trừ khi lỗi được dự tính và xử lý bới lập trình viên, đây là chủ đề nâng cao mà chúng ta sẽ đề cập sau trong khóa này).


Hãy thử chạy code trong cell bên dưới và xem điều gì sẽ xảy ra:


In [7]:
# In string và lỗi để xem thứ tự chạy

print("This will be printed")
frint("This will cause an error")
print("This will NOT be printed")

This will be printed


NameError: ignored

<h3 id="exercise">Exercise: Chương trình đầu tiên</h3>


<p>Nhiều thế hệ lập trình viên đã bắt đầu sự nghiệp viết code của họ bằng cách in dòng chữ "Hello, world!". Bạn sẽ theo bước họ.</p>
<p>Trong cell code bên dưới, sử dụng hàm <code> print()</code> để in ra cụm: <code> Hello, world!</code></p>



In [8]:
# Viết code của bạn xuống dưới đây. Đừng quên ấn Shift+Enter để thực thi cell
print("Hello, world!")

Hello, world!


<details><summary>Click vào đây để xem lời giải</summary>

```python
print("Hello, world!")

```

</details>


<p>Bây giờ, hãy nâng cao code của bạn bằng một chú thích. Trong cell code bên dưới, hãy in ra cụm: <code> Hello, world! </code> và chú thích nó bằng cụm <code> Print the hello world </code> truyền thống, tất cả trong một dòng code.</p>


In [9]:
# Write your code below. Don't forget to press Shift+Enter to execute the cell
print("Hello, world!") # Print the hello world

Hello, world!


<details><summary>Click vào đây để xem lời giải</summary>

```python
print("Hello, world!") # Print the traditional hello world

```

</details>


<hr>


<h2 id="types_objects" align="center">Các kiểu đối tượng trong Python</h2>


<p>Python là ngôn ngữ hướng đối tượng. Có nhiều loại đối tượng khác nhau trong Python. Hãy bắt đầu với các loại đối tượng phổ biến nhất: <i> string (chuỗi) </i>, <i> integer (số nguyên)</i> và <i> float (số thực)</i>. Bất cứ khi nào bạn viết từ (văn bản) bằng Python, bạn đang sử dụng <i> chuỗi ký tự </i> (viết tắt là chuỗi). Mặt khác, các số phổ biến nhất là <i> số nguyên </i> (ví dụ: -1, 0, 100) và <i> float </i>, đại diện cho các số thực (ví dụ: 3,14, -42,0).</p>


<a align="center">
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0101EN-SkillsNetwork/labs/Module%201/images/TypesObjects.png" width="600">
</a>


<p>Các cell code sau chứa một số ví dụ.</p>


In [10]:
# Integer

11

11

In [11]:
# Float

2.14

2.14

In [12]:
# String

"Hello, Python 101!"

'Hello, Python 101!'

<p>Bạn có thể yêu cầu Python cho bạn biết kiểu của biểu thức bằng cách sử dụng hàm <code> type()</code> tích hợp sẵn. Bạn sẽ nhận thấy rằng Python đề cập đến các số nguyên là <code> int</code>, số thực là <code> float</code> và các chuỗi ký tự là <code>str</code>.</p>


In [13]:
# Kiểu của 12

type(12)

int

In [14]:
# Kiểu của 2.14

type(2.14)

float

In [15]:
# Kiểu của "Hello, Python 101!"

type("Hello, Python 101!")

str


<p>Trong cell code bên dưới, sử dụng hàm <code> type()</code> để kiểm tra kiểu đối tượng của <code>12.0</code>.


In [16]:
# Write your code below. Don't forget to press Shift+Enter to execute the cell


<details><summary>Click vào đây để xem lời giải</summary>

```python
type(12.0)

```

</details>


<h3 id="int">Số nguyên</h3>


<p>Dưới đây là một số ví dụ về số nguyên. Số nguyên có thể là số âm hoặc số dương:</p>


<a align="center">
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0101EN-SkillsNetwork/labs/Module%201/images/TypesInt.png" width="600">
</a>


<p> Chúng ta có thể xác minh đây là trường hợp bằng cách sử dụng, chắc bạn cũng đoán được, là hàm <code>type()</code>:


In [17]:
# In ra kiểu của -1

type(-1)

int

In [18]:
# In ra kiểu của 4

type(4)

int

In [19]:
# In ra kiểu của 0

type(0)

int

<h3 id="float">Float</h3>


<p> Float đại diện cho số thực; chúng là một tập hợp các số nguyên nhưng cũng bao gồm "các số thập phân". Có một số hạn chế khi nói đến máy biểu diễn số thực, nhưng số dấu phẩy động là một đại diện tốt trong hầu hết các trường hợp. Bạn có thể tìm hiểu chi tiết thêm về float cho môi trường runtime, bằng cách kiểm tra giá trị của <code> sys.float_info</code>. Điều này cũng sẽ cho bạn biết số lớn nhất và nhỏ nhất có thể được biểu thị với chúng là gì.</p>

<p>Một lần nữa, có thể kiểm tra một số ví dụ với hàm <code> type()</code>:


In [20]:
# In ra kiểu của 1.0

type(1.0) # Notice that 1 is an int, and 1.0 is a float

float

In [21]:
# In ra kiểu của 0.5

type(0.5)

float

In [22]:
# In ra kiểu của 0.56

type(0.56)

float

In [23]:
# Thiết lập hệ thống về kiểu của float

sys.float_info

sys.float_info(max=1.7976931348623157e+308, max_exp=1024, max_10_exp=308, min=2.2250738585072014e-308, min_exp=-1021, min_10_exp=-307, dig=15, mant_dig=53, epsilon=2.220446049250313e-16, radix=2, rounds=1)

<h3 id="convert">Chuyển đổi một kiểu đối tượng sang kiểu đối tượng khác</h3>


<p> Bạn có thể thay đổi kiểu đối tượng trong Python; điều này được gọi là typecasting (ép kiểu). Ví dụ: bạn có thể chuyển đổi <i> số nguyên </i> thành <i> số thực </i> (ví dụ: 2 thành 2.0). </p>
<p> Hãy thử nó: </p>


In [24]:
# Xác thực đây là một số nguyên

type(2)

int

<h4>Chuyển số nguyên thành số thực</h4>
<p>Hãy chuyển số nguyên 2 thành số thực:</p>


In [25]:
# Chuyển 2 thành một số thực

float(2)

2.0

In [26]:
# Chuyển số nguyên 2 thành một số thực và kiểm tra kiểu của nó

type(float(2))

float


<p> Khi chúng ta chuyển đổi một số nguyên thành một số thực, chúng ta không thực sự thay đổi giá trị (tức là ý nghĩa) của số đó. Tuy nhiên, nếu chúng ta ép kiểu số thực thành một số nguyên, chúng ta có thể mất một số thông tin. Ví dụ: nếu chúng ta ép kiểu số thực 1.1 thành số nguyên, chúng ta sẽ nhận được 1 và mất thông tin phần thập phân (tức là 0,1): </p>



In [27]:
# Ép kiểu 1.1 thành số nguyên sẽ mất thông tin

int(1.1)

1

<h4> Chuyển đổi từ string thành số nguyên hoặc số thực </h4>


<p> Đôi khi, chúng ta có thể có một string chứa một số bên trong nó. Nếu đúng như vậy, chúng ta có thể ép string biểu thị một số thành một số nguyên bằng cách sử dụng <code> int()</code>: </p>


In [28]:
# Chuyển string thành một số nguyên

int('1')

1

<p> Nhưng nếu bạn cố gắng làm như vậy với một string không phải là một kết hợp hoàn hảo cho một số, bạn sẽ gặp lỗi. Hãy thử những cách sau: </p>


In [29]:
# Chuyển string thành một số nguyên với lỗi
int('1 or 2 people')

ValueError: ignored

<p> Bạn cũng có thể chuyển đổi các string chứa số dấu phẩy động thành các đối tượng <i> số thực</i>: </p>


In [30]:
# Chuyển đổi string "1.2" thành một số thực

float('1.2')

1.2

<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
    [Tip:] Lưu ý rằng các string có thể được biểu diễn bằng các dấu nháy đơn (<code>'1.2'</code>) hoặc dấu nháy đôi (<code>"1.2"</code>), nhưng bạn không thể kết hợp cả hai (ví dụ, <code>"1.2'</code>).
</div>
<hr/>


<h4>Chuyển số thành string</h4>



<p>Nếu chúng ta có thể chuyển đổi string thành số, thì chúng ta cũng có thể chuyển đổi số thành string phải không?</p>


In [31]:
# Chuyển đổi số nguyên thành string

str(1)


'1'

<p>Và không có lý do gì chúng ta không thể biến số thực thành string:</p>


In [32]:
# Chuyển đổi thực thành string

str(1.2)

'1.2'

<h3 id="bool">Kiểu dữ liệu Boolean</h3>



<p> <i> Boolean </i> là một kiểu dữ liệu quan trọng khác trong Python. Đối tượng kiểu <i> Boolean </i> có thể nhận một trong hai giá trị <code> True</code> hoặc <code> False</code>: </p>


In [33]:
# Giá trị True

True

True

<p> Lưu ý rằng giá trị <code> True </code> có chữ "T" viết hoa. Điều này cũng đúng với <code> False </code> (tức là bạn phải sử dụng chữ "F" viết hoa). </p>


In [34]:
# Giá trị False

False

False


<p> Khi bạn yêu cầu Python hiển thị kiểu đối tượng Boolean, nó sẽ hiển thị <code> bool</code> là viết tắt của <i> boolean</i>: </p>


In [35]:
# Kiểu của True

type(True)

bool

In [36]:
# Kiểu của False

type(False)

bool

<p> Chúng ta có thể chuyển đối tượng boolean sang các kiểu dữ liệu khác. Nếu chúng ta chuyển kiểu boolean có giá trị <code> True</code> thành một số nguyên hoặc số thực, chúng ta sẽ nhận được 1. Nếu chuyển kiểu boolean có giá trị <code> False </code> thành một số nguyên hoặc số thực, chúng ta sẽ nhận được 0. Tương tự, nếu chúng ta chuyển 1 thành Boolean, bạn sẽ nhận được <code> True</code>. Và nếu chúng ta chuyển 0 thành Boolean, chúng ta sẽ nhận được <code> False</code>. Hãy thử xem: </p>


In [37]:
# Chuyển True thành int

int(True)

1

In [38]:
# Chuyển 1 thành boolean

bool(1)

True

In [39]:
# Chuyển 0 thành boolean

bool(0)

False

In [40]:
# Chuyển True thành float

float(True)

1.0

<h3 id="exer_type">Exercise: Các kiểu dữ liệu</h3>


<p>Kiểu dữ liệu của kết quả sau là: <code>6 / 2</code>?</p>


In [41]:
# Viêt code của bạn xuống dưới đây. Đừng quên ấn Shift+Enter để thực thi code


<details><summary>Click vào đây để xem lời giải</summary>

```python
type(6/2) # float

```

</details>


<p>Kiểu dữ liệu của kết quả sau là: <code>6 // 2</code>? (Chú ý dấu <code>//</code>.)</p>


In [42]:
# Viêt code của bạn xuống dưới đây. Đừng quên ấn Shift+Enter để thực thi code


<details><summary>Click vào đây để xem lời giải</summary>

```python
type(6//2) # int, as the double slashes stand for integer division

```

</details>


<hr>


<h2 id="expressions">Biểu thức và biến</h2>


<h3 id="exp">Biểu thức</h3>


<p> Các biểu thức trong Python có thể gồm các phép toán của các kiểu tương thích (ví dụ: số nguyên và số thực). Ví dụ: các phép toán số học cơ bản như cộng nhiều số: </p>

In [43]:
# Biểu thức phép cộng

43 + 60 + 16 + 41

160


<p> Chúng ta có thể thực hiện các phép tính trừ bằng toán tử trừ. Trong trường hợp này, kết quả là một số âm: </p>


In [44]:
# Biểu thức phép trừ

50 - 60

-10

<p>Chúng ta có thể thực hiện phép nhân sử dụng dấu hoa thị: </p>

In [45]:
# Biểu thức phép nhân

5 * 5

25

<p>Chúng ta cũng có thể thực hiện phép chia bằng dấu gạch chéo:


In [46]:
# Biểu thức phép chia

25 / 5

5.0

In [47]:
# Biểu thức phép chia

25 / 6

4.166666666666667


<p> Như đã thấy ở quiz ở trên, chúng ta có thể sử dụng dấu gạch chéo kép để chia số nguyên, trong đó kết quả được làm tròn tới số nguyên gần nhất:

In [48]:
# Biểu thức phép chia số nguyên

25 // 5

5

In [49]:
# Biểu thức phép chia số nguyên

25 // 6

4

<h3 id="exer_exp">Exercise: Biểu thức</h3>



<p> Hãy viết biểu thức tính xem có bao nhiêu giờ trong 160 phút:

In [50]:
# Viêt code của bạn xuống dưới đây. Đừng quên ấn Shift+Enter để thực thi code


<details><summary>Click vào đây để xem lời giải</summary>

```python
160/60

# Or

160//60

```

</details>



<p> Python tuân theo các quy ước toán học được thống nhất khi xét các biểu thức toán học. Trong ví dụ sau, Python thêm 30 vào kết quả của phép nhân (tức là 120).


In [51]:
# Biểu thức toán học

30 + 2 * 60

150


<p> Và cũng giống như toán học, các biểu thức được đặt trong dấu ngoặc đơn được ưu tiên thực hiện trước. Vì vậy, sau đây nhân 32 với 60.


In [52]:
# Biểu thức toán học

(30 + 2) * 60

1920

<h3 id="var">Biến</h3>



<p> Cũng giống như với hầu hết các ngôn ngữ lập trình, chúng ta có thể lưu trữ các giá trị trong <i> biến</i> để có thể sử dụng chúng sau này. Ví dụ: </p>

In [53]:
# Lưu trữ giá trị trong biến

x = 43 + 60 + 16 + 41


<p> Để xem giá trị của <code> x</code> trong Notebook, chúng ta chỉ cần đặt nó vào dòng cuối cùng của cell: </p>


In [54]:
# In ra giá trị trong biến

x

160

<p> Chúng ta cũng có thể thực hiện các phép toán trên <code> x</code> và lưu kết quả vào một biến mới: </p>


In [55]:
# Sử dụng một biến khác để lưu kết quả của phép toán giữa biến và giá trị

y = x / 60
y

2.6666666666666665

<p> Nếu chúng ta lưu giá trị vào một biến hiện có, giá trị mới sẽ ghi đè lên giá trị trước đó: </p>


In [56]:
# Ghi đè biến với giá trị mới

x = x / 60
x

2.6666666666666665

<p> Có một cách hay là sử dụng các tên biến có ý nghĩa để bạn và những người khác có thể đọc và hiểu code dễ dàng hơn: </p>


In [57]:
# Đặt tên biến có ý nghĩa

total_min = 43 + 42 + 57 # Tổng thời gian của album tính theo phút
total_min

142

In [58]:
# Đặt tên biến có ý nghĩa

total_hours = total_min / 60 # Tổng thời gian của album tính theo giờ
total_hours

2.3666666666666667


<p> Trong các cell ở trên, chúng ta đã thêm thời lượng của 3 album tính bằng phút và lưu trữ nó trong <code> total_min</code>. Sau đó, chúng ta chia nó cho 60 để tính tổng thời lượng <code> total_hours </code> theo giờ. Bạn cũng có thể thực hiện tất cả cùng một lúc trong một biểu thức, miễn là bạn sử dụng dấu ngoặc đơn để thêm độ dài album trước khi chia, như minh họa bên dưới. </p>


In [59]:
# Biểu thức phức tạp

total_hours = (43 + 42 + 57) / 60  # Tổng số giờ trong một biểu thức duy nhất
total_hours

2.3666666666666667


<p> Nếu bạn muốn có tổng số giờ dưới dạng số nguyên, tất nhiên bạn có thể thay thế phép chia dấu phẩy động bằng phép chia số nguyên (tức là <code> //</code>). </p>


<h3 id="exer_exp_var">Exercise: Biểu thức và Biến trong Python</h3>


<p>Giá trị của <code>x</code> trong <code>x = 3 + 2 * 2</code> là gì?</p>


In [60]:
# Viêt code của bạn xuống dưới đây. Đừng quên ấn Shift+Enter để thực thi code
x = 3 + 2 * 2
x

7

<details><summary>Click vào đây để xem lời giải</summary>

```python
7

```

</details>


<p>Giá trị của <code>y</code> trong <code>y = (3 + 2) * 2</code> là gì?</p>


In [61]:
# Viêt code của bạn xuống dưới đây. Đừng quên ấn Shift+Enter để thực thi code
y = (3 + 2) * 2
y

10

<details><summary>Click vào đây để xem lời giải</summary>

```python
10

```

</details>


<p>Giá trị của <code>z</code> trong <code>z = x + y</code> là gì?</p>


In [62]:
# Viêt code của bạn xuống dưới đây. Đừng quên ấn Shift+Enter để thực thi code
z = x + y
z

17

<details><summary>Click vào đây để xem lời giải</summary>

```python
17

```

</details>
